### PR TITLE
(DOCSP-34287) Add documentation for `App.EmailPassword.retryCustomConfirmation`

### PR DESCRIPTION
**Pull request by krollins-mdb**
## Pull Request Info

Jira ticket: https://jira.mongodb.org/browse/DOCSP-34287
Staged changes:

- [Manage Email/Password Users](https://preview-mongodbkrollinsmdb.gatsbyjs.io/realm/DOCSP-34457/sdk/dotnet/manage-users/manage-email-password-users/#retry-a-user-confirmation-function)

Add an example for the new App.EmailPassword.retryCustomConfirmation method.

### Reminder Checklist

Before merging your PR, make sure to check a few things.

- [x] Describe your PR's changes in the Release Notes section
- [x] Create a Jira ticket for corresponding docs-app-services work if needed

### Release Notes

- **.NET SDK**
  - Manage Users/Manage Email/Password Users: Add an example for the new App.EmailPassword.retryCustomConfirmation method and the older, undocumented, App.EmailPassword.ResendConfirmationEmailAsync method.

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/dotnet/manage-users/manage-email-password-users.txt [hunk 1]
```diff
@@ -51,6 +51,17 @@ mehtod.
 .. literalinclude:: /examples/generated/dotnet/WorkWithRealm.snippet.confirm-user.cs
    :language: csharp
 
+Retry a User Confirmation Function
```
Comment (cbullinger) [inline source/sdk/dotnet/manage-users/manage-email-password-users.txt:54]: Do we want to also include a section for resending the email a la [Swift](https://www.mongodb.com/docs/realm/sdk/swift/users/manage-email-password-users/#resend-a-user-confirmation-email) -- maybe under another PR? It looks like it's an available [method](https://www.mongodb.com/docs/realm-sdks/dotnet/latest/reference/Realms.Sync.App.EmailPasswordClient.html#Realms_Sync_App_EmailPasswordClient_ResendConfirmationEmailAsync_System_String_) in .NET

File: source/sdk/dotnet/manage-users/manage-email-password-users.txt [hunk 2]
```diff
@@ -51,6 +51,17 @@ mehtod.
...
+----------------------------------
+
+To resend the confirmation email to a user, call the
+:dotnet-sdk:`RetryCustomConfirmationAsync() 
+<reference/Realms.Sync.App.EmailPasswordClient.html#Realms_Sync_App_EmailPasswordClient_RetryCustomConfirmationAsync_System_String_>`
+method with the user's email address.
+
+.. literalinclude:: /examples/generated/dotnet/WorkWithRealm.snippet.retry-custom-confirmation.cs
+   :language: csharp
+
```
Comment (cbullinger) [inline source/sdk/dotnet/manage-users/manage-email-password-users.txt:64]: Link out to the App Svcs docs?
```suggestion

For more information, refer to :ref:`<auth-run-a-confirmation-function>` in the App Services documentation.

```

**Review by cbullinger — APPROVED** [review]
Question for related content that's out-of-scope for this ticket, but otherwise LGTM!

File: source/sdk/dotnet/manage-users/manage-email-password-users.txt (hunk 1 above)
Comment (krollins-mdb) [reply source/sdk/dotnet/manage-users/manage-email-password-users.txt:54]: I'll look at adding it. Shouldn't take too long.

Comment (krollins-mdb) [reply source/sdk/dotnet/manage-users/manage-email-password-users.txt:65]: Ah, yep. I'll update that now.

**Pull request merged**